src/climatevision/generator/industry2018/energy_source/__init__.py

```diff
@@ -4,109 +4,151 @@
-    )
-    fossil_fueloil = EnergyWithPercentage(
-        energy=entries.i_fueloil_fec, total_energy=total_energy_supply
-    )
-    fossil_lpg = EnergyWithPercentage(
-        energy=entries.i_lpg_fec, total_energy=total_energy_supply
-    )
-    fossil_opetpro = EnergyWithPercentage(
-        energy=entries.i_opetpro_fec, total_energy=total_energy_supply
-    )
-    fossil_ofossil = EnergyWithPercentage(
-        energy=entries.i_ofossil_fec, total_energy=total_energy_supply
-    )
-
-    fossil = Energy(
-        energy=fossil_gas.energy
-        + fossil_coal.energy
-        + fossil_diesel.energy
-        + fossil_fueloil.energy
-        + fossil_lpg.energy
-        + fossil_opetpro.energy
-        + fossil_ofossil.energy
-    )
-
-    renew_biomass = EnergyWithPercentage(
-        energy=entries.i_biomass_fec, total_energy=total_energy_supply
-    )
-    renew_heatnet = EnergyWithPercentage(
-        energy=entries.i_heatnet_fec, total_energy=total_energy_supply
-    )
-    renew_heatpump = EnergyWithPercentage(
-        energy=entries.i_orenew_fec * fact("Fact_R_S_ratio_heatpump_to_orenew_2018"),
-        total_energy=total_energy_supply,
-    )
-    renew_solarth = EnergyWithPercentage(
-        energy=entries.i_orenew_fec * fact("Fact_R_S_ratio_solarth_to_orenew_2018"),
-        total_energy=total_energy_supply,
-    )
-    renew_elec = EnergyWithPercentage(
-        energy=entries.i_elec_fec, total_energy=total_energy_supply
-    )
-
-    renew = Energy(
-        energy=renew_biomass.energy
-        + renew_heatnet.energy
-        + renew_heatpump.energy
-        + renew_solarth.energy
-        + renew_elec.energy
-    )
-
-    total = EnergyWithPercentage(
-        energy=entries.i_energy_total, total_energy=entries.i_energy_total
-    )
-
-    return EnergySupply(
-        total=total,
-        fossil=fossil,
-        fossil_gas=fossil_gas,
-        fossil_coal=fossil_coal,
-        fossil_diesel=fossil_diesel,
-        fossil_fueloil=fossil_fueloil,
-        fossil_lpg=fossil_lpg,
-        fossil_opetpro=fossil_opetpro,
-        fossil_ofossil=fossil_ofossil,
-        renew=renew,
-        renew_biomass=renew_biomass,
-        renew_heatnet=renew_heatnet,
-        renew_heatpump=renew_heatpump,
-        renew_solarth=renew_solarth,
-        renew_elec=renew_elec,
+class EnergySource:
+    s: EnergyWithPercentage
+    s_fossil: Energy
+    s_fossil_gas: EnergyWithPercentage
+    s_fossil_coal: EnergyWithPercentage
+    s_fossil_diesel: EnergyWithPercentage
+    s_fossil_fueloil: EnergyWithPercentage
+    s_fossil_lpg: EnergyWithPercentage
+    s_fossil_opetpro: EnergyWithPercentage
+    s_fossil_ofossil: EnergyWithPercentage
+    s_renew: Energy
+    s_renew_biomass: EnergyWithPercentage
+    s_renew_heatnet: EnergyWithPercentage
+    s_renew_elec: EnergyWithPercentage
+    s_renew_orenew: EnergyWithPercentage
+
+
+# TODO Energy calculation with new facts for each sector
```

Comment:
@JanLK You have done this right? That is just an outdated comment is it not?

Reply:
Yes this is done and the comment is outdated.